exchange/bidder.go

```diff
@@ -136,6 +136,9 @@ func (bidder *bidderAdapter) requestBid(ctx context.Context, bidderRequest Bidde
 	//check if real request exists for this bidder or it only has stored responses
 	dataLen := 0
 	if len(bidderRequest.BidRequest.Imp) > 0 {
+		// Reducing the amount of time bidders have to compensate for the processing time used by PBS to fetch a stored request (if needed), validate the OpenRTB request and split it into multiple requests sanitized for each bidder.
+		// As well as for the time needed by PBS to prepare the auction response
+		bidderRequest.BidRequest.TMax = getBidderTmax(&bidderTmaxCtx{ctx}, bidderRequest.BidRequest.TMax, reqInfo.TmaxAdjustments)
```

Comment:
In pull request #2787 a `tmaxAdjustments.BidderResponseDurationMin > 0` check was added line 539 of `exchange/bidder.go` that makes perfect sense given that there's no point in adjusting the `bidderTmax` if `bidder_response_duration_min_ms` wasn't set.  Maybe we could also avoid the `getBidderTmax()` call in the corner case of a `tmax_adjustments` config not making sense if we further validate the other fields of `TmaxAdjustments`. Take the unlikely but feasible config below:
```
  tmax_adjustments:
    enabled: true
    bidder_network_latency_buffer_ms: 0
    pbs_response_preparation_duration_ms: 0
    bidder_response_duration_min_ms: 0
```
If a hypothetical `TmaxAdjustments` check in `config/config.go` or somewhere else upstream sets the `TmaxAdjustments.Enabled` to `false`, or even the entire `TmaxAdjustments` struct to `nil`, we could avoid the `getBidderTmax()` call entirely:
```
138     if len(bidderRequest.BidRequest.Imp) > 0 {
139         // Reducing the amount of time bidders have to compensate for the processing time used by PBS to fetch a stored request (if needed), validate the OpenRTB request and split it into multiple requests sanitized for each bidder.
140         // As well as for the time needed by PBS to prepare the auction response
    +       if reqInfo.TmaxAdjustments != nil {
141             bidderRequest.BidRequest.TMax = getBidderTmax(&bidderTmaxCtx{ctx}, bidderRequest.BidRequest.TMax, reqInfo.TmaxAdjustments)
    +       }
142         reqData, errs = bidder.Bidder.MakeRequests(bidderRequest.BidRequest, reqInfo)
exchange/bidder.go
```
Thoughts?

Reply:
@guscarreon , makes sense to add check for avoid `getBidderTmax()`. https://github.com/prebid/prebid-server/pull/2803/commits/870e0582a0dc24c6433dcc07f7622c3dc4332b7c makes these change.

also https://github.com/prebid/prebid-server/pull/2803/commits/8bd2d16ec72bf914ffe5b0115b9fa80188f0ac5d makes change to not subtract `PBSResponsePreparationDuration`. Because Deadline context subtracts PBSResponsePreparationDuration in `makeAuctionContext()`. So no need to subtract it again

https://github.com/prebid/prebid-server/blob/d54d22b9266b2104ebbc35a586e5774ef5f61547/exchange/exchange.go#L632-L649